lib/src/application_configuration.dart

```diff
@@ -0,0 +1,79 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:pub_semver/pub_semver.dart';
+
+enum MetadataPersistenceMode {
+  plainText,
+  encrypted,
+}
```

Comment:
Missing docs

Reply:
We will user `unencrypted` instead of `plainText`. Also, added forgotten `disabled` value, and added doc comments